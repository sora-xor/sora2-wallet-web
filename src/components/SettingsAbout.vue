<template>
  <wallet-base :title="t('settings.menu.About.title')" show-back @back="handleBack">
    <div class="wallet-settings-about">
      <span class="wallet-settings-about_desc">{{ appInfo }}</span>
      <div
        v-for="(link, index) in links"
        :key="index"
      >
        <div class="wallet-settings-about-item s-flex">
          <div class="wallet-settings-about-item-text s-flex">
            <a class="wallet-settings-about-item-text_main" :href="link.url" target="_blank">{{ link.title }}</a>
          </div>
          <!-- This link was hidden due to PSS-205 task. We'll return it back later.  -->
          <!-- <s-icon
            class="wallet-settings-about-item_icon"
            name="external-link"
            :size="16"
          /> -->
        </div>
        <s-divider v-if="index !== links.length - 1" class="wallet-settings-about-item_divider" />
      </div>
    </div>
  </wallet-base>
</template>

<script lang="ts">
import { Component, Mixins, Prop } from 'vue-property-decorator'
import { Action, State } from 'vuex-class'

import TranslationMixin from './mixins/TranslationMixin'
import WalletBase from './WalletBase.vue'
import { RouteNames } from '../consts'

@Component({
  components: {
    WalletBase
  }
})
export default class SettingsAbout extends Mixins(TranslationMixin) {
  readonly links = [
    {
      title: this.t('settings.about.termsOfService'),
      url: this.t('settings.about.termsOfServiceLink')
    },
    {
      title: this.t('settings.about.privacyPolicy'),
      url: this.t('settings.about.privacyPolicyLink')
    }
  ]

  @Action navigate

  get appInfo (): string {
    const year = new Date().getFullYear()
    return `Sora2 Wallet ${year}`
  }

  handleBack (): void {
    this.navigate({ name: RouteNames.WalletSettings })
  }
}
</script>

<style scoped lang="scss">
.wallet-settings-about {
  &_desc {
    @include hint-text;
    padding: $basic-spacing_small 0;
  }
  &-item {
    justify-content: space-between;
    padding: $basic-spacing_small 0;
    &-text {
      flex-direction: column;
      &_main {
        text-decoration: unset;
        color: unset;
        font-size: $font-size_normal;
        line-height: $line-height_medium;
      }
    }
    &_icon {
      margin: auto 0;
    }
    &_divider {
      margin: unset;
    }
    &:hover {
      cursor: pointer;
      .wallet-settings-about-item-text_main,
      .wallet-settings-about-item_icon {
        color: var(--s-color-button-tertiary-color);
      }
    }
  }
}
</style>
