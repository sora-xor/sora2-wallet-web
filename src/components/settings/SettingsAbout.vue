<template>
  <wallet-base :title="t('settings.menu.About.title')" show-back @back="handleBack">
    <div class="wallet-settings-about">
      <span class="wallet-settings-about_desc">{{ appInfo }}</span>
      <div
        v-for="(link, index) in links"
        :key="index"
      >
        <div class="wallet-settings-about-item s-flex">
          <div class="wallet-settings-about-item-text s-flex">
            <a class="wallet-settings-about-item-text_main" :href="link.url" target="_blank">{{ link.title }}</a>
          </div>
          <s-icon
            class="wallet-settings-about-item_icon"
            name="external-link"
            :size="16"
          />
        </div>
        <s-divider v-if="index !== links.length - 1" style="margin: unset" />
      </div>
    </div>
  </wallet-base>
</template>

<script lang="ts">
import { Component, Mixins, Prop } from 'vue-property-decorator'
import { Action, State } from 'vuex-class'

import TranslationMixin from '../mixins/TranslationMixin'
import WalletBase from '../WalletBase.vue'
import { RouteNames } from '../../consts'

@Component({
  components: {
    WalletBase
  }
})
export default class SettingsAbout extends Mixins(TranslationMixin) {
  readonly links = [
    {
      title: 'Terms of Service',
      url: 'https://sora.org'
    },
    {
      title: 'Privacy Policy',
      url: 'https://soramitsu.co.jp'
    }
  ]

  @Action navigate

  get appInfo () {
    const year = new Date().getFullYear()
    return `Sora Neo Wallet ${year}.`
  }

  handleBack (): void {
    this.navigate({ name: RouteNames.WalletSettings })
  }
}
</script>

<style scoped lang="scss">
@import '../../styles/typography';
@import '../../styles/layout';
@import '../../styles/soramitsu-variables';

.wallet-settings-about {
  &_desc {
    color: $s-color-base-content-tertiary;
    font-size: $font-size_small;
    padding: $basic-spacing_small 0;
  }
  &-item {
    justify-content: space-between;
    padding: $basic-spacing_small 0;
    &-text {
      flex-direction: column;

      &_main {
        text-decoration: unset;
        color: unset;
        font-size: $font-size_normal;
        line-height: 1.8;
      }
    }

    &_icon {
      margin: auto 0;
    }

    &:hover {
      cursor: pointer;
      .wallet-settings-about-item-text_main,
      .wallet-settings-about-item_icon {
        color: $s-color-button-tertiary-color;
      }
    }
  }
}
</style>
