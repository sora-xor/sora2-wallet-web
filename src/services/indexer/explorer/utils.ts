import { formatStringNumber } from '../../../util';

import type { SubqueryAssetEntity } from '../subquery/types';
import type { SubsquidAssetEntity } from '../subsquid/types';
import type { FiatPriceObject, UpdatesStream, PoolApyObject } from '../types';

export function parseAssetFiatPrice(entity: SubsquidAssetEntity | SubqueryAssetEntity): FiatPriceObject {
  const acc = {};
  const id = entity.id;
  const priceFPNumber = formatStringNumber(entity.priceUSD);
  const isPriceFinity = priceFPNumber.isFinity();
  if (isPriceFinity) {
    acc[id] = priceFPNumber.toCodecString();
  }
  return acc;
}

export function parsePriceStreamUpdate(entity: UpdatesStream): Nullable<FiatPriceObject> {
  if (!entity?.data) return null;

  const data = JSON.parse(entity.data);

  return Object.entries(data).reduce((acc, [id, price]) => {
    const priceFPNumber = formatStringNumber(price as string);
    const isPriceFinity = priceFPNumber.isFinity();
    if (isPriceFinity) {
      acc[id] = priceFPNumber.toCodecString();
    }
    return acc;
  }, {});
}

export function parseApyStreamUpdate(entity: UpdatesStream): PoolApyObject {
  const data = entity?.data ? JSON.parse(entity.data) : {};

  return Object.entries(data).reduce((acc, [id, apy]) => {
    const strategicBonusApyFPNumber = formatStringNumber(apy as string);
    const isStrategicBonusApyFinity = strategicBonusApyFPNumber.isFinity();
    if (isStrategicBonusApyFinity) {
      acc[id] = strategicBonusApyFPNumber.toCodecString();
    }
    return acc;
  }, {});
}
