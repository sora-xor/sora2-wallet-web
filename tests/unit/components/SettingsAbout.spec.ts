import Vuex from 'vuex'
import { shallowMount, createLocalVue } from '@vue/test-utils'

import SettingsAbout from '@/components/SettingsAbout.vue'
import { TranslationMock, SoramitsuElementsImport } from '../../utils'

const localVue = createLocalVue()
localVue.use(Vuex)
const store = new Vuex.Store({
  modules: {
    Router: {
      actions: {
        navigate: jest.fn()
      }
    }
  } as any
})

describe('SettingsAbout.vue', () => {
  beforeEach(() => {
    SoramitsuElementsImport(localVue)
    TranslationMock(SettingsAbout)
  })

  it('should renders correctly', () => {
    const wrapper = shallowMount(SettingsAbout, { localVue, store })
    expect(wrapper.element).toMatchSnapshot()
  })
})
