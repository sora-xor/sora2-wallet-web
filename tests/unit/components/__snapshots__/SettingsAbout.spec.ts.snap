// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SettingsAbout.vue should renders correctly 1`] = `
<wallet-base-stub
  showback="true"
  title=""
>
  <div
    class="wallet-settings-about"
  >
    <span
      class="wallet-settings-about_desc"
    >
      SORA Wallet
    </span>
     
    <div>
      <div
        class="wallet-settings-about-item s-flex"
      >
        <div
          class="wallet-settings-about-item-text s-flex"
        >
          <a
            class="wallet-settings-about-item-text_main"
            target="_blank"
          >
            
          </a>
        </div>
      </div>
       
      <s-divider-stub
        class="wallet-settings-about-item_divider"
        contentposition="center"
        direction="horizontal"
        type="secondary"
      />
    </div>
    <div>
      <div
        class="wallet-settings-about-item s-flex"
      >
        <div
          class="wallet-settings-about-item-text s-flex"
        >
          <a
            class="wallet-settings-about-item-text_main"
            target="_blank"
          >
            
          </a>
        </div>
      </div>
       
      <!---->
    </div>
  </div>
</wallet-base-stub>
`;
